--- mkimage-yum.sh.orig	2018-01-05 11:35:00.087100742 +0900
+++ mkimage-yum.sh	2018-01-05 16:21:23.291785420 +0900
@@ -18,6 +18,7 @@
                    The default is "Core".
   -y <yumconf>     The path to the yum config to install packages from. The
                    default is /etc/yum.conf for Centos/RHEL and /etc/dnf/dnf.conf for Fedora
+  -v               set vault.centos.org in /etc/yum.repos.d/CentOS-Base.repo
 EOOPTS
     exit 1
 }
@@ -29,7 +30,8 @@
 	alias yum=dnf
 fi
 install_groups="Core"
-while getopts ":y:p:g:h" opt; do
+vault=0
+while getopts ":y:p:g:hv" opt; do
     case $opt in
         y)
             yum_config=$OPTARG
@@ -43,6 +45,9 @@
         g)
             install_groups="$OPTARG"
             ;;
+		v)
+			vault=1
+			;;
         \?)
             echo "Invalid option: -$OPTARG"
             usage
@@ -87,10 +92,22 @@
 if [[ -n "$install_packages" ]];
 then
     yum -c "$yum_config" --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
-        --setopt=group_package_types=mandatory -y install "$install_packages"
+        --setopt=group_package_types=mandatory -y install $install_packages
 fi
 
-yum -c "$yum_config" --installroot="$target" -y clean all
+# Fix up RPM database
+rm "$target"/var/lib/rpm/*
+chroot "$target" /bin/rpm --initdb
+chroot "$target" /bin/rpm -i --justdb '/var/cache/yum/*/*/*/packages/*.rpm'
+chroot "$target" /usr/bin/yum -y clean all
+
+if [ $vault = 1 ] ; then
+	version=$(grep '^baseurl=http://vault.centos.org/' $yum_config | head -n 1 | cut -d/ -f 4)
+	sed -i \
+		-e '/^mirrorlist=/N;s/^mirrorlist=[^\n]*\n//' \
+		-e 's|^#baseurl=http://mirror.centos.org/centos/$releasever/|baseurl=http://vault.centos.org/'"$version"'/|' \
+		"$target"/etc/yum.repos.d/CentOS-Base.repo
+fi
 
 cat > "$target"/etc/sysconfig/network <<EOF
 NETWORKING=yes
